plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/expression/ImplementCount.java

```diff
@@ -57,7 +58,9 @@ public ImplementCount(JdbcTypeHandle bigintTypeHandle)
     @Override
     public Pattern<AggregateFunction> getPattern()
     {
-        return basicAggregation()
+        return Pattern.typeOf(AggregateFunction.class)
+                .with(hasSortOrder().equalTo(false))
+                .with(hasFilter().equalTo(false))
```

Comment:
That should consider what is the type of the input argument and look out for varchars.

Add a test with `count(DISTINCT c)` over table with `A, a, b, B, a`.